services/route_manager/src/main/java/com/futurewei/alcor/route/controller/NeutronRouterController.java

```diff
@@ -347,12 +370,29 @@ public RoutesToNeutronWebResponse removeRoutesToNeutronRouter(@PathVariable Stri
         RestPreconditionsUtil.verifyParameterNotNullorEmpty(routerid);
         RestPreconditionsUtil.verifyResourceFound(projectid);
 
-        RoutesToNeutronRouterRequestObject router = resource.getRouter();
-        if (router == null) {
+        NewRoutesWebRequest newRoutes = resource.getRouter();
+        if (newRoutes == null) {
             return new RoutesToNeutronWebResponse();
         }
+        // List<String> ports -> port entity -> subnet id
+        Router router = this.routerDatabaseService.getByRouterId(routerid);
+
+        RoutesToNeutronWebResponse routesToNeutronWebResponse = this.neutronRouterService.removeRoutesToNeutronRouter(routerid, newRoutes);
+
+        List<String> gatewayPorts = router.getPorts();
+        List<String> subnetIds = this.routerToPMService.getSubnetIdsFromPM(projectid, gatewayPorts);
+        // sub-level routing rule update
+        List<InternalSubnetRoutingTable> internalSubnetRoutingTableList = new ArrayList<>();
+        for (String subnetId : subnetIds) {
+            UpdateRoutingRuleResponse updateRoutingRuleResponse = this.neutronRouterService.updateRoutingRule(subnetId, newRoutes, true);
+            List<InternalSubnetRoutingTable> internalSubnetRoutingTables = updateRoutingRuleResponse.getInternalSubnetRoutingTables();
+            internalSubnetRoutingTableList.addAll(internalSubnetRoutingTables);
```

Comment:
same comment as in the `addRoutesToNeutronRouter`

Reply:
fixed it